src/frontend/src/optimizer/plan_node/logical_multi_join.rs

```diff
@@ -727,6 +690,79 @@ impl LogicalMultiJoin {
     }
 }
 
+// Join tree internal representation
+#[derive(Clone, Default, Debug)]
+struct JoinTreeNode {
+    idx: Option<usize>,
+    left: Option<Box<JoinTreeNode>>,
+    right: Option<Box<JoinTreeNode>>,
+    height: usize,
+}
+
+// join graph internal representation
+#[derive(Clone, Debug)]
+struct GraphNode {
+    id: usize,
+    join_tree: JoinTreeNode,
+    // use BTreeSet for deterministic
+    relations: BTreeSet<usize>,
+}
+
+///  equivalent condition derivation by `a = b && a = c` ==> `b = c`
+fn eq_condition_derivation(mut nodes: BTreeMap<usize, GraphNode>) -> BTreeMap<usize, GraphNode> {
```

Comment:
In addition to adding edges between graph nodes, we need to add the corresponding equal conditions.